### PR TITLE
Fixing #4981

### PR DESCRIPTION
**Pull request by sanjana-bhat**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/KubernetesListOperationsImpl.java [hunk 1]
```diff
@@ -57,6 +58,9 @@ public ClientKubernetesListNonNamespaceOperation inNamespace(String namespace) {
   @Override
   public KubernetesList create(KubernetesList... items) {
     List<HasMetadata> createdItems = new ArrayList<>();
+    if (items.length == 0) {
+      items[0] = get();
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/KubernetesListOperationsImpl.java]: This isn't going to work. You'll need to do something like `items = new KubernetesList[]{get()};`

**Comment by jimmidyson** [comment]
Thanks for the PR - minor fix needed. You should be able to run tests locally with `mvn clean install` before submitting - that way you can see the test failures locally & fix up before submitting.

**Comment by sanjana-bhat** [comment]
@jimmidyson and @iocanel, thanks for reviewing it. I'm still not able to run tests locally. All the tests fail due to config parsing failure. Do I need to include a kubeconfig and if so in what path? Any env variables that need to be set? Thanks for all your help, really appreciate it.

**Comment by jimmidyson** [comment]
Are you running Windows by any chance? Are the failures the same as #226?

**Comment by sanjana-bhat** [comment]
Nope. I tried running on both Mac OS and RHEL7
Errors on Mac OS (no k8s running)
```
io.fabric8.kubernetes.client.KubernetesClientException: Error executing: POST at: https://mybox:57384/api/v1/namespaces/test/services. Cause: connect timed out
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:345)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at com.squareup.okhttp.internal.Platform.connectSocket(Platform.java:100)
	at com.squareup.okhttp.Connection.connectSocket(Connection.java:196)
	at com.squareup.okhttp.Connection.connect(Connection.java:172)
	at com.squareup.okhttp.Connection.connectAndSetOwner(Connection.java:367)
	at com.squareup.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:128)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:328)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:245)
	at com.squareup.okhttp.Call.getResponse(Call.java:267)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:224)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils$3.intercept(HttpClientUtils.java:109)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:221)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:195)
	at com.squareup.okhttp.Call.execute(Call.java:79)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:201)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:183)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:466)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:206)
	at io.fabric8.kubernetes.client.handlers.ServiceHandler.create(ServiceHandler.java:35)
	at io.fabric8.kubernetes.client.handlers.ServiceHandler.create(ServiceHandler.java:25)
	at io.fabric8.kubernetes.client.dsl.internal.KubernetesListOperationsImpl.create(KubernetesListOperationsImpl.java:104)
	at io.fabric8.kubernetes.client.dsl.internal.KubernetesListOperationsImpl.create(KubernetesListOperationsImpl.java:66)
	at io.fabric8.kubernetes.client.dsl.internal.KubernetesListOperationsImpl.create(KubernetesListOperationsImpl.java:40)
	at io.fabric8.kubernetes.client.mock.KubernetesListTest.testCreate(KubernetesListTest.java:54)
```
Errors on RHEL 7 (where I have k8s running)
```
testGet(io.fabric8.kubernetes.client.mock.NodeTest)  Time elapsed: 4.579 sec  <<< ERROR!
java.lang.StackOverflowError: null
	at com.fasterxml.jackson.databind.introspect.AnnotatedClass._collectRelevantAnnotations(AnnotatedClass.java:899)
	at com.fasterxml.jackson.databind.introspect.AnnotatedClass._collectRelevantAnnotations(AnnotatedClass.java:892)
	at com.fasterxml.jackson.databind.introspect.AnnotatedClass._constructConstructor(AnnotatedClass.java:803)
	at com.fasterxml.jackson.databind.introspect.AnnotatedClass.resolveCreators(AnnotatedClass.java:345)
	at com.fasterxml.jackson.databind.introspect.AnnotatedClass.getStaticMethods(AnnotatedClass.java:245)
	at com.fasterxml.jackson.databind.introspect.BasicBeanDescription.getFactoryMethods(BasicBeanDescription.java:461)
	at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory._addDeserializerFactoryMethods(BasicDeserializerFactory.java:670)
	at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory._constructDefaultValueInstantiator(BasicDeserializerFactory.java:321)
	at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory.findValueInstantiator(BasicDeserializerFactory.java:254)
	at com.fasterxml.jackson.databind.deser.BasicDeserializerFactory.createCollectionDeserializer(BasicDeserializerFactory.java:1027)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer2(DeserializerCache.java:394)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:352)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:264)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findNonContextualValueDeserializer(DeserializationContext.java:451)
	at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer._findCustomDeser(UntypedObjectDeserializer.java:162)
	at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer.resolve(UntypedObjectDeserializer.java:134)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:428)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:947)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:517)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:428)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.createContextual(CollectionDeserializer.java:164)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.createContextual(CollectionDeserializer.java:25)
	at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:669)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:430)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:947)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:439)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:428)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:947)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:439)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:428)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.createContextual(CollectionDeserializer.java:164)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.createContextual(CollectionDeserializer.java:25)
	at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:669)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:430)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:947)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:439)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:461)
	at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:3838)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3732)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2620)
	at io.fabric8.kubernetes.client.internal.KubeConfigUtils.parseConfig(KubeConfigUtils.java:40)
	at io.fabric8.kubernetes.client.Config.tryKubeConfig(Config.java:239)
	at io.fabric8.kubernetes.client.Config.<init>(Config.java:106)
	at io.fabric8.kubernetes.client.Config.<init>(Config.java:122)
	at io.fabric8.kubernetes.client.EditableConfig.<init>(EditableConfig.java:12)
	at io.fabric8.kubernetes.client.ConfigBuilder.build(ConfigBuilder.java:30)
	at io.fabric8.kubernetes.client.internal.SSLUtils.isHttpsAvailable(SSLUtils.java:58)
	at io.fabric8.kubernetes.client.Config.<init>(Config.java:112)
	at io.fabric8.kubernetes.client.Config.<init>(Config.java:122)
	at io.fabric8.kubernetes.client.EditableConfig.<init>(EditableConfig.java:12)
	at io.fabric8.kubernetes.client.ConfigBuilder.build(ConfigBuilder.java:30)
	at io.fabric8.kubernetes.client.internal.SSLUtils.isHttpsAvailable(SSLUtils.java:58)
	at io.fabric8.kubernetes.client.Config.<init>(Config.java:112)
	at io.fabric8.kubernetes.client.Config.<init>(Config.java:122)
	at io.fabric8.kubernetes.client.EditableConfig.<init>(EditableConfig.java:12)
	at io.fabric8.kubernetes.client.ConfigBuilder.build(ConfigBuilder.java:30)
	at io.fabric8.kubernetes.client.internal.SSLUtils.isHttpsAvailable(SSLUtils.java:58)
```

**Comment by sanjana-bhat** [comment]
Any prerequisites for tests to be run?

**Comment by jimmidyson** [comment]
No there should be none. You don't need k8s running as we've mocked out all server interaction.

Any ideas @iocanel?

**Comment by jimmidyson** [comment]
@iocanel I guess we should always use a project local kubeconfig file in tests so we don't use any users real one from their home dir?

**Comment by sanjana-bhat** [comment]
@jimmidyson, I don't have any kubeconfig file or env variables on my Mac. I just checked out the project and ran mvn clean install on it.

**Comment by jimmidyson** [comment]
I was more talking about the RHEL7 issue tbh. Would you be able to share your kube config file from RHEL7, obviously with all sensitive info obfuscated (passwords, usernames, etc)? I see multiple issues here that we need to solve.

**Comment by sanjana-bhat** [comment]
This is all I have in my kubeconfig
cat /etc/k8s/kubeconfig
```
apiVersion: v1
clusters:
- cluster:
    server: http://127.0.0.1:8080
contexts:
kind: Config
preferences: {}
```

**Comment by jimmidyson** [comment]
How about in `~/.kube/config`? Anything at all? Have you set `KUBECONFIG` environment variable pointing to any file?

**Comment by sanjana-bhat** [comment]
I haven't set any env variable. Nothing at all in ~/.kube/config

**Comment by jimmidyson** [comment]
Could you rebase against master & try again? Fixed some dodgy logic in Config ctor.

**Comment by sanjana-bhat** [comment]
I'm able to run the tests now on RHEL7. Thanks! I will fix the tests for this PR

**Comment by jimmidyson** [comment]
Ah excellent! Sorry about that. Still need to figure out why it doesn't build for you on OSX but we can do that separately.

**Comment by sanjana-bhat** [comment]
@jimmidyson, I have fixed the tests. Thanks for all your help!

**Comment by jimmidyson** [comment]
Ok to test

**Comment by jimmidyson** [comment]
ok to test

**Comment by jimmidyson** [comment]
[merge]

**Comment by jimmidyson** [comment]
Could you rebase & push again please? Need to sort the merge policy soon as this is a bit annoying but for now need to rebase manually. Sorry!

**Comment by sanjana-bhat** [comment]
Done

**Comment by jimmidyson** [comment]
[merge]

**Comment by jimmidyson** [comment]
Thank you!

**Pull request merged**